script.R

```diff
@@ -88,61 +130,53 @@ options("repos" = c(
+x <- pkgdepends::new_pkg_deps(".")
+x$resolve()
+pkgs_from_gh <- Filter(is_installed_from_gh, unique(x$get_resolution()$package))
+for (pkg in pkgs_from_gh) {
+  add_cache_to_minicran(pkg, installed.packages()[pkg, "Version"], minicran_path)
+}
+
+## revdepcheck
+cli::cli_progress_step("Initiating `revdepcheck`...")
+revdepcheck::revdep_reset()
+unlink("./revdep/", recursive = TRUE)
+revdepcheck:::db_disconnect(".")
+usethis::use_revdep()
+revdepcheck:::db_setup(".")
+
 cli::cli_progress_done()
 
 
 cli::cli_h1("Add refs to revdepcheck")
 # include refs in revdepcheck
-## Add revdep to miniCRAN repo so that it can be found by the revdepcheck.
-## miniCRAN accepts only prebuilt .tar.gz file and build requires all the dependencies pre-installed.
-## This is why we need to install all the deps of revdep (incl. tested package).
-## It's important to use `crancache` as much as possible to make use of caching.
-## Algorithm:
-## for ref in refs:
-## - install dependencies using `crancache`
-## - install pkg using `pak` - this also gives prebuilt .tar.gz file
-## - move .tar.gz file to miniCRAN repo
-## - add to revdep todo table
+## Add refs revdepcheck and also to miniCRAN so that it can be found from there
 cli::cli_progress_bar("Adding refs to revdepcheck", total = length(refs))
 for (ref in refs) {
   cli::cli_progress_message("Adding {ref}...")
 
-  ref_parsed <- pkgdepends::parse_pkg_ref(ref)
-  ref_pkg <- ref_parsed$package
-
-  if (!is(ref_parsed, "remote_ref_standard") && !is(ref_parsed, "remote_ref_cran")) {
-    cli::cli_progress_message("Installing dependencies of {ref}...")
-    ref_deps <- pkgdepends::new_pkg_deps(ref, config = list(dependencies = FALSE))
-    ref_deps$resolve()
-    ref_deps_df <- ref_deps$get_resolution()[1, "deps"][[1]]
-    ref_deps_hard <- ref_deps_df[
-      tolower(ref_deps_df$type) %in% tolower(pkgdepends::pkg_dep_types_hard()) & ref_deps_df$ref != "R",
-      "package"
-    ]
-    crancache::install_packages(ref_deps_hard, quiet = TRUE)
-
-    cli::cli_progress_message("Installing {ref}...")
-    ref_install <- pak::pkg_install(ref)
-    ref_cache <- pkgcache::pkg_cache_find(package = ref_pkg)
-    ref_targz <- subset(
-      ref_cache,
-      (built == 1 | built == TRUE | built == "TRUE") & platform == "source" & version == ref_install$version[1],
-      fullpath
-    )[[1]]
-    # cache might have multiple files for a given package and version
-    # copy this file to the temp dir and add to miniCRAN from that dir
-    temp_dir <- tempfile()
-    dir.create(temp_dir)
-    file.copy(
-      ref_targz,
-      file.path(temp_dir, paste0(sub("(.*?_.*?)_.*", "\\1", basename(ref_targz)), ".tar.gz"))
-    )
-    miniCRAN::addLocalPackage(ref_pkg, temp_dir, minicran_path)
-
-    cli::cli_inform("Added {ref} to minicran!")
-  }
+  install_and_add_to_minicran(ref, minicran_path)
 
+  ref_pkg <- pkgdepends::parse_pkg_ref(ref)$package
   revdepcheck::revdep_add(packages = ref_pkg)
+
```

Comment:
Just a minor thing, but now this lines 

https://github.com/insightsengineering/r-revdepcheck-action/blob/1fc867c2e149a60fb0de86da26ac5c68516a9c14/script.R#L192

gives 

```r
Installing CRAN version of teal.goshawk
```

even though it's a GitHub version

Reply:
Yes. It's hardcoded in revdepcheck and we can't do anything about that :(